test/e2e/tests/settings/ipfs-ens-resolution.spec.js

```diff
@@ -1,190 +1,54 @@
-      return {
-        statusCode: 200,
-        json: {
-          jsonrpc: '2.0',
-          id: (await req.body.getJson()).id,
-          result:
-            '0x0000000000000000000000000000000000000000000000000000000000000000',
-        },
-      };
-    });
-}
 
 describe('Settings', function () {
   const ENS_NAME = 'metamask.eth';
   const ENS_NAME_URL = `https://${ENS_NAME}/`;
   const ENS_DESTINATION_URL = `https://app.ens.domains/name/${ENS_NAME}`;
 
-  it('Redirects to ENS domains when user inputs ENS into address bar', async function () {
-    // Using proxy port that doesn't resolve so that the browser can error out properly
-    // on the ".eth" hostname.  The proxy does too much interference with 8000.
-
+  it(`Redirects to ENS domains when user inputs ENS into address bar`, async function () {
+    async function mockMetaMaskDotEth(mockServer) {
+      return await mockServer.forGet(ENS_NAME_URL).thenResetConnection();
+    }
+    async function mockEnsDotDomains(mockServer) {
+      return await mockServer
+        .forGet(ENS_DESTINATION_URL)
+        .thenReply(200, 'mocked ENS domain');
+    }
+    async function mockEns(mockServer) {
+      return [
+        await mockMetaMaskDotEth(mockServer),
+        await mockEnsDotDomains(mockServer),
+      ];
+    }
     await withFixtures(
       {
-        fixtures: new FixtureBuilder().withNetworkControllerOnMainnet().build(),
-        ganacheOptions: defaultGanacheOptions,
-        title: this.test.title,
-        testSpecificMock: mockInfura,
-        driverOptions: {
-          proxyPort: '8001',
-        },
+        title: this.test.fullTitle(),
+        testSpecificMock: mockEns,
       },
       async ({ driver }) => {
-        await driver.navigate();
+        // if the background/offscreen pages are ready metamask should be ready
+        // to handle ENS domain resolution
+        const page =
+          process.env.ENABLE_MV3 === 'false'
+            ? PAGES.BACKGROUND
+            : PAGES.OFFSCREEN;
+        await driver.navigate(page);
 
         // The setting defaults to "on" so we can simply enter an ENS address
         // into the address bar and listen for address change
         try {
           await driver.openNewPage(ENS_NAME_URL);
         } catch (e) {
-          // Ignore ERR_PROXY_CONNECTION_FAILED error
-          // since all we care about is getting to the correct URL
+          // Ignore ERR_CONNECTION_RESET as it's an intentional return value
+          // by our mocks
+          if (!e.message.includes('ERR_CONNECTION_RESET')) {
+            throw e;
+          }
         }
 
-        // Ensure that the redirect to ENS Domains has happened
-        await driver.wait(async () => {
-          const currentUrl = await driver.getCurrentUrl();
-          return currentUrl === ENS_DESTINATION_URL;
-        }, 10000);
-        // Setting a large delay has proven to stabilize the flakiness of the redirect
-        // and it's only a MAX value
```

Comment:
added a bit more sophisticated function on the driver.js file called `waitForUrl` to handle this logic, so we can re-use it whenever is needed, which also includes a retrial and timeout